step-framework-timeseries/src/main/java/step/core/timeseries/aggregation/TimeSeriesAggregationPipeline.java

```diff
@@ -5,69 +5,243 @@
 import org.slf4j.LoggerFactory;
 import step.core.collections.Collection;
 import step.core.collections.Filter;
+import step.core.timeseries.TimeSeriesCollection;
+import step.core.timeseries.TimeSeriesFilterBuilder;
 import step.core.timeseries.bucket.Bucket;
 import step.core.timeseries.bucket.BucketAttributes;
 import step.core.timeseries.bucket.BucketBuilder;
-import step.core.timeseries.TimeSeriesFilterBuilder;
 
 import java.util.*;
 import java.util.concurrent.atomic.LongAdder;
-import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 
 public class TimeSeriesAggregationPipeline {
 
     private static final Logger logger = LoggerFactory.getLogger(TimeSeriesAggregationPipeline.class);
+    // resolution - array index
+    private final Map<Long, Integer> resolutionsIndexes = new HashMap<>();
+    // sorted
+    private final List<TimeSeriesCollection> collections;
 
-    private final long sourceResolution;
-    private final Collection<Bucket> collection;
-
-    public TimeSeriesAggregationPipeline(Collection<Bucket> collectionDriver, long resolution) {
-        this.collection = collectionDriver;
-        this.sourceResolution = resolution;
+    public TimeSeriesAggregationPipeline(List<TimeSeriesCollection> collections) {
+        this.collections = collections;
+        for (int i = 0; i < collections.size(); i++) {
+            TimeSeriesCollection collection = collections.get(i);
+            resolutionsIndexes.put(collection.getResolution(), i);
+        }
     }
 
-    protected long getSourceResolution() {
-        return sourceResolution;
-    }
+    public TimeSeriesAggregationResponse collect(TimeSeriesAggregationQuery query) {
+        validateQuery(query);
+        long idealResolution = 0;
+        if (query.getOptimizationType() == TimeSeriesOptimizationType.MOST_ACCURATE) {
```

Comment:
Not covered by JUnit. Should be done in a follow-up task

Reply:
we have the task already and is in progress